packages/flutter/lib/src/material/material_state.dart

```diff
@@ -371,26 +368,29 @@ abstract class MaterialStateUnderlineInputBorder extends UnderlineInputBorder im
   /// Creates a [MaterialStateUnderlineInputBorder] from a [MaterialPropertyResolver<InputBorder>]
   /// callback function.
   ///
-  /// If used as a regular input border, the border resolved in the default state (the
-  /// empty set of states) will be used.
+  /// [MaterialStateUnderlineInputBorder] should only be used with widgets that document
+  /// their support, like [InputDecoration.border].
   ///
-  /// The given callback parameter must return a non-null text style in the default
+  /// The given callback parameter must return a non-null input border in the default
   /// state.
   const factory MaterialStateUnderlineInputBorder.resolveWith(MaterialPropertyResolver<InputBorder> callback) = _MaterialStateUnderlineInputBorder;
 
+  /// Creates a [MaterialStateUnderlineInputBorder] from a [WidgetStateMap<InputBorder>].
+  ///
+  /// {@macro flutter.widgets.WidgetStateProperty.fromMap}
+  ///
+  /// [MaterialStateUnderlineInputBorder] should only be used with widgets that document
+  /// their support, like [InputDecoration.border].
+  ///
+  /// {@macro flutter.widgets.WidgetState.any}
+  const factory MaterialStateUnderlineInputBorder.fromMap(WidgetStateMap<InputBorder> map) = _UnderlineInputBorderMapper;
```

Comment:
I sort of forgot that `MaterialStateUnderlineInputBorder` and `MaterialStateOutlineInputBorder` also exist 😅

Reply:
These classes in particular of pretty specific properties make me want union types...